internal/state/dataplane/configuration.go

```diff
@@ -476,12 +488,37 @@ func createFilters(filters []v1beta1.HTTPRouteFilter) Filters {
 	for _, f := range filters {
 		switch f.Type {
 		case v1beta1.HTTPRouteFilterRequestRedirect:
-			result.RequestRedirect = f.RequestRedirect
-			// using the first filter
-			return result
+			if result.RequestRedirect == nil {
+				// using the first filter
+				result.RequestRedirect = f.RequestRedirect
+			}
+		case v1beta1.HTTPRouteFilterRequestHeaderModifier:
+			if result.RequestHeaderModifiers == nil {
+				// using the first filter
+				result.RequestHeaderModifiers = convertHTTPFilter(f.RequestHeaderModifier)
+			}
 		}
 	}
+	return result
+}
 
+func convertHTTPFilter(httpFilter *v1beta1.HTTPHeaderFilter) *HTTPHeaderFilter {
+	result := &HTTPHeaderFilter{}
+	result.Remove = httpFilter.Remove
+	for _, s := range httpFilter.Set {
+		newSet := HTTPHeader{
+			Name:  string(s.Name),
+			Value: s.Value,
+		}
+		result.Set = append(result.Set, newSet)
```

Comment:
nit: the struct is small enough to combine the initialization and append statements 
```suggestion
		result.Set = append(result.Set, HTTPHeader{Name: string(s.Name), Value: s.Value})
```